### PR TITLE
Java library examples

### PR DESCRIPTION
**Pull request by sbesson**
See https://trello.com/c/M7U9L70L/74-document-developer-changes-re-poms

Our current Bio-Formats develope documentation contains two sections providing minimal examples of  how to consume Bio-Formats as a Java library:
- the last paragraph of http://www.openmicroscopy.org/site/support/bio-formats5.3/developers/building-bioformats.html
- the first section of http://www.openmicroscopy.org/site/support/bio-formats5.3/developers/java-library.html

Both paragraphs contain snippets for consuming Bio-Formats using either Maven or Gradle. However, the current layout using the Sphinx documentation has limitations. First, the content is not update (currently using 5.1.0-SNAPSHOT). Second, these example files are never tested in a regular manner. As an example, the requirement to declare the unidata repository for retrieving the `netcdf` artifacts is currently not documented /cc @chris-allan @emilroz 

In order to address these issues, the proposal of this PR is to migrate these minimal examples out of the Sphinx repository into its own GitHub repository and point to it from the documentation instead. To do this, the idea is to resurrect and repurpose this existing repository originally targetted for API examples. The code layout is largely inspired by our own [minimal OMERO client repository](https://github.com/ome/minimal-omero-client) or [the Spock example repository](https://github.com/spockframework/spock-example) repository.

Changes included in this PR are the following:
- migration of the `pom.xml` and `build.gradle` files from the Bio-Formats SPhinx documentation
- upgrade of the components to 5.3.x
- activation of Travis CI for the systematic validation of the Maven and Gradle builds
- update of the Gradle build to use `gradle wrapper` in order to catch build issues
- addition of the Unidata repositories to `pom.xml` and `build.gradle`
- updates the README to describe the purpose of the repository

The proposed follow-up actions would be to:
- rename this repository as `bio-formats-examples` (matching the conventions of all new BF related repositories)
- register the update of this repository as a downstream step following each Bio-Formats release
- update the Bio-Formats Sphinx developer documentation to remove the sections and point at this repository instead

/cc @melissalinkert @jburel @joshmoore @dgault

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,40 @@
...
+
+    JDK 7 or higher
+    Maven 2.x (for Maven build)
+
+Building with Maven
+-------------------
+
+From the root directory, run:
+
+    mvn
```
Comment (dgault) [inline README.md]: The project builds fine when a goal is specified but not by default. Either the doc could be updated to include a suitable goal  (mvn clean install) or a default goal could be added to the pom eg <defaultGoal>install</defaultGoal>

Comment (sbesson) [inline README.md]: :+1: to keep the instructions minimal, f6edfde7baabe1b365ffcfed271df8938893bf9f adds a default goal running the `clean` and `install` phases.

**Comment by dgault** [comment]
With the last commit both the maven and gradle builds work as expected from the documentation. 

Im certainly happy with the layout of the repo and the readme looks and reads fine also.

If this can be used to keep those examples up to date and easier maintained then it looks like a positive step in the right direction from my perspective.

**Comment by hflynn** [comment]
The plan works for me too - much better to move examples to where it's easier to keep them up to date. Readme looks fine too.

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,3 @@
+.gradle
+target
+build
```
Comment (jburel) [inline .gitignore]: add extra line

File: gradlew [hunk 3]
```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/env sh
+
+##############################################################################
+##
+##  Gradle start up script for UN*X
```
Comment (jburel) [inline gradlew]: Do you mean UNIX?

Comment (sbesson) [inline gradlew]: This file (and the corresponding `.bat` one) are generated by the `gradle wrapper` command so I don't think it is worth reviewing their content.

Comment (jburel) [inline gradlew]: understood

**Pull request merged**